runtime/tests/interpreter/member_test.go

```diff
@@ -1197,4 +1197,97 @@ func TestInterpretNestedReferenceMemberAccess(t *testing.T) {
 		_, err := inter.Invoke("test")
 		require.ErrorAs(t, err, &interpreter.InvalidMemberReferenceError{})
 	})
+
+	t.Run("struct entitlement escalation", func(t *testing.T) {
+		t.Parallel()
+
+		inter := parseCheckAndInterpret(t, `
+
+            entitlement E
+
+            struct T {
+                access(E) fun foo() {}
+            }
+
+            struct S {
+                access(mapping Identity) var ref: AnyStruct
+        
+                init(_ a: AnyStruct){
+                    self.ref = a
+                }
+            }
+        
+            fun test() {
+                let t = T()
+                let pubTRef = &t as &T
+                var s = &S(pubTRef) as auth(E) &S
+                var tRef = s.ref as! auth(E) &T
```

Comment:
Can you add a test for the same, but instead of `pubTRef = &t as &T`, with an entitled reference; `pubTRef = &t as auth(E) &T`?

I guess that should pass?
i.e: doing `s.ref as! auth(E) &T` in two steps as
```cadence
var member = s.ref
var tRef =  member as! auth(E) &T
```
already works? (not sure)

Reply:
No, this also must fail. Here the static type of `member` is `&AnyStruct` after the `s.ref` access. We cannot allow this to be downcast to `auth(E) &T` because entitlements are erased at runtime, so there is no way to regain the `E` entitlement after it is lost. I added a test for this case though.

Reply:
ah right, `member` is `&AnyStruct` (not `AnyStruct` which is what I was thinking) 👍 . Makes sense